### PR TITLE
feat(react-blessed): add props from blessed, react, local defs

### PR DESCRIPTION
**Pull request by defnotrobbie**
# pr template

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

## existing definition update

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

previous `react-blessed` JSX definitions can be narrowed using `blessed` and `react` definitions.

### CODE REVIEW

**Comment by defnotrobbie** [comment]
i've apparently misunderstood how `jsx-runtime` works. the project i was testing the updated defs with uses babel to compile TS and doesn't have an issue with a new `jsx-runtime` in `tsconfig.json` (i guess because babel config doesn't use it). i created a simple TS-only react/react-blessed project and the compiled code expects an actual `jsx-runtime` script in `react-blessed`, not just alternate JSX definitions.

**Review by orta — APPROVED** [review]

**Comment by defnotrobbie** [comment]
@orta Thanks for the review! I'm going fix a few typos in code comments and merge.

**Comment by defnotrobbie** [comment]
Ready to merge

**Comment by defnotrobbie** [comment]
> @orta Thanks for the review! I'm going fix a few typos in code comments and merge.

ahh, this triggered a re-review requirement. :/ @orta would you mind re-reviewing? there aren't any code changes, just comments.

File: types/react-blessed/tsconfig.json
Comment (defnotrobbie) [reply types/react-blessed/tsconfig.json:18]: @elibarzilay i guess not--though i believed the bumped min TS version will always work with this (the new jsx transform)?

will remove.

**Comment by defnotrobbie** [comment]
no idea what's going on with these errors unless there were breaking changes to TS? i've plopped the definitions and test in typescript playground running TS4.3 beta and none of these errors show up.

**Comment by elibarzilay** [comment]
@defnotrobbie: sorry about the noise -- it was not clear what should be done with the `jsx` option, which was bad since I was debugging the bot when you were trying to get it to run. I now made the bot ignore that option, so you can try reverting the change to it which might have caused the test failure.  (It should consistently ignore the option now.)

**Comment by defnotrobbie** [comment]
ABANDONED! I THINK NOT!

builds passed for over a month while this PR was awaiting review then the CI changed and no more build passes?

**Comment by orta** [comment]
I tried rebasing and pushing a new commit to this PR, but it's marked as not letting upstream contributors  have access to push. The CI fails do look like they were legit (they came from typescript improvements in master in-between your commits) so it's likely they're still legit fails now

I'm afraid this has been idle for so long because PRs Failing CI don't end up on the DT maintainers TODO list (given the scale of how many PRs are on this repo) - so my recommendation if you still want to see these changes happen are to give it a rebase and check out how it runs on CI now

**Comment by defnotrobbie** [comment]
> I tried rebasing and pushing a new commit to this PR, but it's marked as not letting upstream contributors have access to push. The CI fails do look like they were legit (they came from typescript improvements in master in-between your commits) so it's likely they're still legit fails now
> 
> I'm afraid this has been idle for so long because PRs Failing CI don't end up on the DT maintainers TODO list (given the scale of how many PRs are on this repo) - so my recommendation if you still want to see these changes happen are to give it a rebase and check out how it runs on CI now

ah i see. that makes sense. i'm glad some type errors were caught before this got merged. will rebase when i have time (unless closing this and opening a new PR later is preferable?).

**Comment by orta** [comment]
You take your time, every time we comment the bot basically adds 2 weeks to the timeout

**Review by orta — APPROVED** [review]
Cool, thanks for the ping!

**Comment by defnotrobbie** [comment]
Ready to merge

**Pull request merged**
